lib/cylc/rundb.py

```diff
@@ -1009,3 +1039,566 @@ def upgrade_pickle_to_json(self):
     def vacuum(self):
         """Vacuum to the database."""
         return self.connect().execute("VACUUM")
+
+
+class CylcReviewDAO(object):
```

Comment:
I can see the following arguments as to whether this would be better as an individual module:

* Cylc Review is independent from suite runtime scheduling & therefore should really be placed in a separate file to the central suite runtime DAO.
* ``rundb.py``, with ``CylcReviewDAO`` included, is now too long (>1500 lines);

however...

* Both ``CylcSuiteDAO`` & ``CylcReviewDAO`` are database access objects & hence have similar purposes, use similar modules, & contain similar methods & attributes (some/most of which we might eventually want to unify e.g. as shared module variables & functions after a more detailed audit).
* ``CylcReviewDAO`` queries a ``CylcSuiteDAO`` DB, hence is closely tied to that class.

@cylc/core: thoughts, opinions?

Reply:
(There is already a another module that deals with runtime database stuff - `cylc.suite_db_mgr`. Makes sense to separate new class in separate module so a suite run does not have to load `cylc review` stuff and vice versa.)

Reply:
Fair enough, I will create a new module for it.